mem/mem.cpp

```diff
@@ -276,29 +278,44 @@ void mem_free_sub(void *memblock) {
     free(memblock);
   }
 }
+
 void mem_error_msg(const char *file, int line, int size) {
   mprintf(0, "Memory error (size=%d) line %d in %s\n", size, line, file);
   Int3();
 }
+
 char *mem_strdup_sub(const char *string, const char *file, int line) {
   char *ret = strdup(string);
   if (!ret) {
     mprintf(0, "Out of memory allocating %d bytes: line %d in %s\n", strlen(string) + 1, line, file);
     Int3();
-    return NULL;
+    return nullptr;
   }
   return ret;
 }
+
 void *mem_realloc_sub(void *mem, int size) { return realloc(mem, size); }
+
 int mem_size_sub(void *memblock) {
-#if defined(MACOSX)
+#ifdef __LINUX__
+#ifdef MACOSX
   return malloc_size(memblock);
 #else
   return malloc_usable_size(memblock);
 #endif
+#elif WIN32
+  return _msize(memblock);
+#else
```

Comment:
we can safely assume that platform is either LINUX/MACOS or WIN32, it's already what we do in many other places

Reply:
Actually, there is `ifdef` on global level for platform check, so this part of code actually unreachable. Removed.